internal/transport/http2_client.go

```diff
@@ -137,11 +137,26 @@ type http2Client struct {
 	connectionID uint64
 }
 
-func dial(ctx context.Context, fn func(context.Context, string) (net.Conn, error), addr string) (net.Conn, error) {
-	if fn != nil {
-		return fn(ctx, addr)
+func dial(ctx context.Context, fn func(context.Context, string) (net.Conn, error), addr resolver.Address, useProxy bool, grpcUA string) (net.Conn, error) {
+	if fn == nil {
```

Comment:
Per @menghanl's previous comments, you don't need to operate on `fn` in this function; do the operations that are needed inline instead, or call `fn` by itself.  Don't assign anything new to `fn`.

Reply:
No longer setting anything to fn (just calling it immediately and returning if it is nil, so proxy wrapper won't be used with custom dialer). Only reason not to change newProxyDialer to directly do default dialing behavior: the tests for it appear to use custom dialer behavior, so those tests would break.

Reply:
Made this change.